ntp-client/src/Network/NTP/Client.hs

```diff
@@ -58,10 +58,10 @@ data NtpClientSettings = NtpClientSettings
 data NtpClient = NtpClient
     { ntpGetStatus        :: IO NtpStatus -- Use STM here ?
       -- ^ Query the current status (non-blocking).
-    , ntpActNow           :: IO ()
+    , ntpForceCheck           :: IO ()
```

Comment:
Shouldn't `ntpForceCheck` return `NtpStatus`? Otherwise there's no way to know that it has already finished (`ntpGetStatus` only allows polling which is not performant).  Since you have the `TVar NtpStatus` at hand you can await untill `NtpStatus` is not `NtpSyncPending`:

```
await :: TVar NtpStatus -> IO NtpStatus
await v = atomically $ do
   s <- readTVar v
   when (s == NtpSyncPending) retry
   pure s
```

Reply:
Yes but lets do that later and first merge the easy stuff, i.e. renaming the package etc.

Reply:
ok